projects/kit/components/slider/helpers/slider-key-steps.directive.ts

```diff
@@ -1,24 +1,59 @@
-import {Directive, forwardRef, inject, Input} from '@angular/core';
+import {
+    Directive,
+    forwardRef,
+    // eslint-disable-next-line no-restricted-syntax
+    HostBinding,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import type {TuiValueTransformer} from '@taiga-ui/cdk/classes';
 import {TuiControl} from '@taiga-ui/cdk/classes';
 import {tuiFallbackValueProvider} from '@taiga-ui/cdk/tokens';
-import {tuiClamp} from '@taiga-ui/cdk/utils/math';
 
 import {TuiSliderComponent} from '../slider.component';
 import type {TuiKeySteps} from './key-steps';
-import {tuiKeyStepValueToPercentage, tuiPercentageToKeyStepValue} from './key-steps';
+import {tuiCreateKeyStepsTransformer} from './key-steps';
 
 @Directive({
     standalone: true,
     selector: 'input[tuiSlider][keySteps]',
+})
+export class TuiSliderKeyStepsBase {
+    private readonly slider = inject<TuiSliderComponent>(
+        forwardRef(() => TuiSliderComponent),
+    );
+
+    // eslint-disable-next-line no-restricted-syntax
+    @HostBinding('attr.aria-valuemin')
+    protected min: number | null = this.slider.min();
+
+    // eslint-disable-next-line no-restricted-syntax
+    @HostBinding('attr.aria-valuemax')
+    protected max: number | null = this.slider.max();
+
+    public transformer = signal<TuiValueTransformer<number, number> | null>(null);
+
+    @Input()
+    public set keySteps(steps: TuiKeySteps | null) {
+        this.transformer.set(steps && tuiCreateKeyStepsTransformer(steps, this.slider));
+        this.min = steps?.[0][1] ?? null;
```

Comment:
Unnecessary fallback if we allow it to be `undefined`? We only use it for aria, right? So we can probably do that